pkg/api/persistentvolumeclaim/util.go

```diff
@@ -29,10 +29,18 @@ const (
 
 // DropDisabledFields removes disabled fields from the pvc spec.
 // This should be called from PrepareForCreate/PrepareForUpdate for all resources containing a pvc spec.
-func DropDisabledFields(pvcSpec, oldPVCSpec *core.PersistentVolumeClaimSpec) {
+func DropDisabledFields(pvc, oldPVC *core.PersistentVolumeClaim) {
+	pvcSpec := &pvc.Spec
+	var oldPVCSpec *core.PersistentVolumeClaimSpec
+	if oldPVC != nil {
+		oldPVCSpec = &oldPVC.Spec
+	}
 	if !dataSourceIsEnabled(pvcSpec) && !dataSourceInUse(oldPVCSpec) {
 		pvcSpec.DataSource = nil
 	}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.RecoverVolumeExpansionFailure) && !allocatedResourcesInUse(oldPVC) {
```

Comment:
In create, we clear status first. In update, we set new status to old status. I think we want this logic in the status strategy instead path instead.

Reply:
hmm, we do not have a createStrategy for pvc status though. We could add one but I have not seen any other example which sets status field via statusstrategy on create.

Reply:
ah. I see what you mean. yeah I could add a new function for status update strategy.

Reply:
I moved it.